### PR TITLE
Runtime regularisation

### PR DESCRIPTION
**Pull request by Krever**
Introduces a common interface for runtimes, more regular creation and adds documentation.

### CODE REVIEW

File: website/docs/other/production-readiness.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Production Readiness
+
+Workflows4s has started a Proof of Concept validating the approach and feasibility. While its
```
Comment (greshny) [inline website/docs/other/production-readiness.md]: should it be something like:
`Workflows4s has started a Proof of Concept validating the approach and feasibility. While its initial stages have shown promising results, the following criteria must be met to ensure production readiness:`

File: workflows4s-pekko/src/main/scala/workflows4s/runtime/pekko/PekkoRuntime.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  def createInstance(id: String): WorkflowInstance[Future, WCState[Ctx]]
+  def initializeShard(): Unit
+}
+
+class PekkoRuntimeImpl[Ctx <: WorkflowContext, Input <: WCState[Ctx]](
+    workflow: Initial[Ctx, Input],
+    initialState: EntityContext[?] => Input,
+    entityName: String,
+    clock: Clock,
+)(implicit
```
Comment (greshny) [inline workflows4s-pekko/src/main/scala/workflows4s/runtime/pekko/PekkoRuntime.scala]: perhaps `using` would fit here

**Review by greshny — APPROVED** [review]
Looks great!

File: website/docs/other/production-readiness.md (hunk 1 above)
Comment (Krever) [reply website/docs/other/production-readiness.md]: Yeah, I completely forgot I havent finished it. Thanks for spotting.

**Review by greshny — APPROVED** [review]

**Pull request merged**